buy_together_app/views.py

```diff
@@ -7,3 +9,34 @@ def main_page(request):
 
 def description_page(request):
     return render(request, 'buy_together_app/description.html')
+
+
+def signup_user(request, msg=""):
+    form = SupplierForm()
+    formm = ClientForm()
+    context = {'form': form, 'formm': formm, 'message': [msg]}
+    return render(request, 'buy_together_app/sign_up.html', context)
+
+
+def signup_supplier(request):
+    if request.method == 'POST':
+        form = SupplierForm(request.POST)
+        if form.is_valid():
+            try:
+                form.save()
+                return redirect('Main Page')
```

Comment:
Which line should raise the exception? Can we take the other line out of the try block?

Reply:
Fixed!